.github/workflows/main.yml

```diff
@@ -361,6 +361,56 @@ jobs:
           name: 'obs-studio-windows-${{ matrix.arch }}-${{ steps.setup.outputs.commitHash }}'
           path: '${{ env.FILE_NAME }}'
 
+  source_build:
+    name: '02 - Source'
+    runs-on: [ubuntu-22.04]
+    if: always()
+    needs: [clang_check]
+    defaults:
+      run:
+        shell: bash
+        working-directory: 'obs-studio'
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+          path: 'obs-studio'
+          fetch-depth: 0
+
+      - name: 'Setup environment'
+        id: setup
+        run: |
+          echo "::set-output name=commitHash::$(git rev-parse --short=9 HEAD)"
+
+      - name: 'Create source artifact'
+        if: ${{ success() && github.event_name != 'pull_request' }}
+        run: |
+          outdir=$(pwd)/build
+          mkdir -p ${outdir}
+          prefix=obs-studio-${{ steps.setup.outputs.commitHash }}
+          outfile=${prefix}.tar
+          rm -f ${outfile}
+          # exclude repository configuration files
+          find .git -type d -name info -exec touch {}/attributes ";"
+          echo ".git* export-ignore" | find .git -type f -path "*/info/attributes" -exec tee -a {} +
+          # create a tarball with all the submodules
+          # this uses uncompressed 'tar', so we can append the archives
+          git archive --format tar -o ${outdir}/${outfile}  --prefix=$prefix/ HEAD
+          PREFIX=${prefix} OUTFILE=${outdir}/${outfile} git submodule --quiet foreach --recursive 'git archive --format tar -o submodule.tar --prefix=$PREFIX/$displaypath/ $sha1; tar -Af ${OUTFILE} submodule.tar; rm -f submodule.tar'
+          # zip and upload
+          gzip ${outdir}/${outfile}
+          ARTIFACT_NAME=$(/usr/bin/find ${outdir} -maxdepth 1 -type f -name "${outfile}.*" -exec basename {} ";" -quit)
```

Comment:
Given that you create a single file with a pre-defined filename, you should not need to use `find` here. `-exec` is also.. not great (see the requirement to add `";"` or `\;` after it to be able to use `-quit`), but not a blocker.

Reply:
that would rely on `gzip` adding a `.gz` prefix (which of course it does).

in any case, i mimicked the behaviour of the existing jobs, that already uses `find` for what seems to be pre-defined filenames as well, see https://github.com/obsproject/obs-studio/blob/dc30cf0843c54f3f02ad412daa10038784ae3be1/.github/workflows/main.yml#L192

i personally find the use of `-exec` with `";"` and `-quit` more clean and resilient than the existing use of piping the output of `find` into `head`.